### PR TITLE
add null check to gmail_query

### LINKED ISSUE #79: delete_message fails due to NULL response
Per the google API documentation regarding `DELETE`:

> If successful, this method returns an empty response body.

This causes `gmail_query` to fail, as it is expecting a non-empty response

**Example**
```library(gmailr)

use_secret_file("email-automation.json")

msgs <- messages(label_ids = "Label_14")
local_path <- "."

for (ids in msgs[[1]]$messages) {
  id <- ids$id
  msg <- message(id)
  save_attachments(msg, path = local_path)
  delete_message(id)
}
```

**Error**
`Error in class(result) <- class : attempt to set an attribute on NULL`

### PR DESCRIPTION
**Pull request by josibake**
check for null value in result. this keeps gmailr_DELETE from failing. closes #79

### CODE REVIEW

File: R/gmailr.R [hunk 1]
```diff
@@ -339,7 +339,7 @@ gmailr_query <- function(fun, location, user_id, class = NULL, upload = FALSE,
     stop(cond)
   }
 
-  if (!is.null(class)) {
+  if (!is.null(class) & !is.null(result)) {
```
Comment (jimhester) [inline R/gmailr.R]: Can you use `&&` instead of `&`. The latter is vectorized, which isn't appropriate here.

**Comment by jimhester** [comment]
Thanks! I had a minor code comment, and can you also add a note to NEWS.md with the fix and referencing the issue # and your github handle.

**Comment by josibake** [comment]
ah! great catch with &&. thanks! I added a bullet under `0.7.1`. is that the right place for adding notes? @jimhester

**Comment by jimhester** [comment]
Yeah there is fine, I can reformat before the next release of gmailr. Thanks again!

**Pull request merged**